arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/ReactorStreamConsumer.java

```diff
@@ -0,0 +1,65 @@
+package io.arex.inst.dynamic.common.listener;
+
+import io.arex.agent.bootstrap.ctx.TraceTransmitter;
+import io.arex.inst.dynamic.common.DynamicClassExtractor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class ReactorStreamConsumer {
+
+    private final TraceTransmitter traceTransmitter;
+    private final DynamicClassExtractor extractor;
+
+    public ReactorStreamConsumer(DynamicClassExtractor extractor) {
+        this.traceTransmitter = TraceTransmitter.create();
+        this.extractor = extractor;
+    }
+
+    /**
+     * support for Mono type recording
+     * @param responseMono
+     * @return
+     */
+    public Mono<?> monoAccept(Mono<?> responseMono) {
+        return responseMono
+            .doOnSuccess(result -> {
+                traceTransmitter.transmit();
+                extractor.recordResponse(result);
+
+            })
+            .doOnError(error-> {
+                traceTransmitter.transmit();
+                extractor.recordResponse(error);
+            });
+    }
+
+
+    /**
+     * support for Flux type recording
+     * @param responseFlux
+     * @return
+     */
+    public Flux<?> fluxAccept(Flux<?> responseFlux) {
+        AtomicBoolean doOnError = new AtomicBoolean(false);
+        final Object[] error = new Object[1];
+        List<Object> fluxList = new ArrayList<>();
+
+        return responseFlux
+            .doOnNext(fluxList::add)
+            .doOnError(err -> {
+                error[0] = err;
+                doOnError.set(true);
+            })
+            .doFinally(result -> {
+                traceTransmitter.transmit();
+                if (doOnError.get()) {
+                    extractor.recordResponse(error[0]);
+                } else {
+                    extractor.recordResponse(fluxList.toArray());
+                }
+            });
+    }
```

Comment:
A Reactive Streams [Publisher](https://www.reactive-streams.org/reactive-streams-1.0.3-javadoc/org/reactivestreams/Publisher.html?is-external=true) with rx operators that emits 0 to N elements, and then completes (successfully or with an error).